[CLD-276]: fix(aptos): migrate to env.BlockChains.AptosChains

graham-chainlink opened the pull request:
Instead of env.AptosChains, we are migrating to env.BlockChains.AptosChains() as part of the newer design refactoring.

Also updated `cldf.AptosChain`  to `cldf_aptos.Chain` as that is the new location

JIRA: https://smartcontract-it.atlassian.net/browse/CLD-276

graham-chainlink committed:
fix(aptos): migrate to blockchain field

Instead of env.AptosChains, we are migrating to env.BlockChains.AptosChains() as part of the newer design refactoring.

JIRA: https://smartcontract-it.atlassian.net/browse/CLD-276

jkongie commented on core/scripts/go.mod:
```diff
@@ -38,7 +38,7 @@ require (
 	github.com/smartcontractkit/chainlink-common v0.7.1-0.20250522183927-44d96a7ad0e5
 	github.com/smartcontractkit/chainlink-data-streams v0.1.1-0.20250417193446-eeb0a7d1e049
-	github.com/smartcontractkit/chainlink-deployments-framework v0.5.0
+	github.com/smartcontractkit/chainlink-deployments-framework v0.5.1-0.20250527010109-bbcfc3657004
```
Do you want to release a 5.1 instead?

graham-chainlink replied:
yeah i do actually, i was waiting for my other PR to go in, it just went in, i will perform a new release now

jkongie commented on deployment/common/changeset/test_helpers.go:
```diff
@@ -138,30 +137,19 @@ func ApplyChangesets(t *testing.T, e cldf.Environment, timelockContractsPerChain
-			blockChains[selector] = ch
-		}
-		for selector, ch := range e.SolChains {
-			blockChains[selector] = ch
```
This seems to remove all assignment to the blockchains. Is that correct?

graham-chainlink replied:
yeah turns out these are not needed as we are just cloning the cldf.Env, so i can just use the existing e.BlockChains.

jkongie commented on deployment/common/changeset/test_helpers.go:
```diff
@@ -214,17 +202,6 @@ func ApplyChangesetsV2(t *testing.T, e cldf.Environment, changesetApplications [
 			// do nothing, as these jobs auto-accept.
 		}
 
-		blockChains := map[uint64]chain.BlockChain{}
-		for selector, ch := range e.Chains {
-			blockChains[selector] = ch
-		}
-		for selector, ch := range e.SolChains {
-			blockChains[selector] = ch
-		}
-		for selector, ch := range e.AptosChains {
-			blockChains[selector] = ch
-		}
-
```
Same here

jkongie commented on deployment/environment/memory/environment.go:
```diff
@@ -328,7 +328,7 @@ func NewMemoryEnvironment(t *testing.T, lggr logger.Logger, logLevel zapcore.Lev
 		chains,
 		solChains,
-		aptosChains,
+		nil, // this field will be deleted in future
```
we should update this comment to explain why it will be deleted so we give more context

graham-chainlink replied:
Updated!

graham-chainlink committed:
fix(aptos): migrate to new aptos packge

Instead of cldf.AptosChain which is an alias, we will now import cldf_aptos.Chain directly

JIRA: https://smartcontract-it.atlassian.net/browse/CLD-276

graham-chainlink committed:
update gomod
